finality-aleph/src/import.rs

```diff
@@ -55,8 +55,17 @@ where
         block: BlockImportParams<Block, Self::Transaction>,
     ) -> Result<ImportResult, Self::Error> {
         let post_hash = block.post_hash();
+
+        self.metrics.convert_header_hash_to_post_hash(
+            block.header.hash(),
+            post_hash,
+            Checkpoint::Importing,
+        );
+
+        // Self-created blocks are imported without using the import queue,
+        // so we need to report them here.
         self.metrics
-            .report_block(post_hash, Instant::now(), Checkpoint::Importing);
+            .report_block_if_not_present(post_hash, Instant::now(), Checkpoint::Importing);
```

Comment:
Are you absolutely sure this is the only place that needs the `if_not_present` check? I think this is the case, but it's annoyingly fragile, so I would like to make sure.

Reply:
Good point, I checked and we also have to check presence when we receive proposal (this was previously a bug) and when adding to import queue,